### PR TITLE
Lv2 ports

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Partly Resolves: #2978

Separate immutable descriptions of the various kinds of LV2 ports
from structures that hold associated state.

Just one of these is the EffectSettings, and as with Ladspa, it is just a
vector<float>.  Except -- will also transiently carries Lilv_state after loading a preset.

The other state structures will later belong to instance or validator objects.

Also some source file splitting for levelization.

The misleadingly named "LV2EffectSettingsDialog" is also renamed and moved into a separate file. 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/effects/lv2/LV2Effect.cpp [hunk 1]
```diff
@@ -1395,6 +1395,8 @@ std::unique_ptr<LV2Wrapper> LV2Effect::InitInstance(float sampleRate)
    wrapper->SetBlockSize();
    wrapper->SetSampleRate();
 
+   static float blackHole;
```
Comment (crsib) [inline src/effects/lv2/LV2Effect.cpp:1284]: Well, that's a name

File: src/effects/lv2/LV2Effect.cpp [hunk 2]
```diff
@@ -99,12 +102,13 @@ BEGIN_EVENT_TABLE(LV2EffectMeter, wxWindow)
...
 END_EVENT_TABLE()
 
-LV2EffectMeter::LV2EffectMeter(wxWindow *parent, const LV2ControlPortPtr port)
-:  wxWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxDEFAULT_CONTROL_BORDER),
-   mControlPort(port)
+LV2EffectMeter::LV2EffectMeter(
+   wxWindow *parent, const LV2ControlPortPtr port, const float &value
+)  : wxWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
+      wxDEFAULT_CONTROL_BORDER), mControlPort(port)
+   , mValue{ value }
```
Comment (crsib) [inline src/effects/lv2/LV2Effect.cpp]: We can drop `const float& value` I think and just initialize `mValue{ port->mVal }` probably.

I've spent some time to convince myself that the lifetime of the pointers (references) match.  But I haven't seen next commits, probably they justify the need for the third argument.

Comment (crsib) [reply src/effects/lv2/LV2Effect.cpp]: Yep, they do justify

**Review by crsib — APPROVED** [review]
I do enjoy the direction LV2 is going

File: src/effects/lv2/LV2Effect.cpp [hunk 3]
```diff
@@ -1755,7 +1755,7 @@ bool LV2Effect::BuildPlain(EffectSettingsAccess &access)
          for (auto & p : mGroupMap[label]) { auto & port = mControlPorts[p];
             auto labelText = port->mName;
             if (!port->mUnits.empty())
-               labelText += wxT(" (") + port->mUnits + wxT(")");
+               labelText += XO("(%s)").Format(port->mUnits).Translation();
```
[the remark this answers is not in the document]
Comment (Paul-Licameli) [reply src/effects/lv2/LV2Effect.cpp:1432]: The string XO(“%s”) occurs elsewhere with i18n comments.

Even punctuation may need to localize.  For instance the Chinese locales prefer slightly different parenthesis characters that are more rounded.

File: src/effects/lv2/LV2Effect.cpp (hunk 1 above)
Comment (Paul-Licameli) [reply src/effects/lv2/LV2Effect.cpp:1112]: Haha

Comment (Paul-Licameli) [reply src/effects/lv2/LV2Effect.cpp:1112]: I could have called it `bitBucket`

File: src/effects/lv2/LV2Effect.cpp (hunk 2 above)
Comment (Paul-Licameli) [reply src/effects/lv2/LV2Effect.cpp]: But, as you must understand by now, this leads up to removal of `mVal` from the port structure so it too is an immutable

File: src/effects/lv2/LV2Effect.h [hunk 4]
```diff
@@ -227,6 +227,15 @@ class LV2ControlPort : public LV2Port
    std::vector<double> mScaleValues;
    wxArrayString mScaleLabels;
 
+   //! Map a real number to one of the scale points
+   size_t Discretize(float value) const {
```
[the remark this answers is not in the document]
Comment (Paul-Licameli) [reply src/effects/lv2/LV2Effect.h]: I did that in the final version

**Pull request merged**